packages/@n8n/task-runner-python/src/task_executor.py

```diff
@@ -272,34 +307,27 @@ def _extract_json_data_per_item(user_output):
         return user_output
 
     @staticmethod
-    def _put_result(
-        queue: multiprocessing.Queue, result: list[Any], print_args: PrintArgs
-    ):
-        json_bytes = json.dumps(result, default=str, ensure_ascii=False).encode("utf-8")
-        json_bytes_size = len(json_bytes)
-
-        shm = shared_memory.SharedMemory(create=True, size=json_bytes_size)
-        shm.buf[:json_bytes_size] = json_bytes
-
-        print_args_to_send = TaskExecutor._truncate_print_args(print_args)
-        queue.put(
-            {
-                "shm_name": shm.name,
-                "shm_size": json_bytes_size,  # stay exact, shm.size can round up for alignment
-                "print_args": print_args_to_send,
-            }
-        )
+    def _put_result(write_fd: int, result: Items, print_args: PrintArgs):
+        message: PipeResultMessage = {
+            "result": result,
+            "print_args": TaskExecutor._truncate_print_args(print_args),
+        }
+
+        data = json.dumps(message, default=str, ensure_ascii=False).encode("utf-8")
+        length_bytes = len(data).to_bytes(PIPE_MSG_PREFIX_LENGTH, "big")
 
-        shm.close()
+        TaskExecutor._write_bytes(write_fd, length_bytes)
+        TaskExecutor._write_bytes(write_fd, data)
+        os.close(write_fd)
 
     @staticmethod
     def _put_error(
-        queue: multiprocessing.Queue,
+        write_fd: int,
         e: BaseException,
         stderr: str = "",
         print_args: PrintArgs = [],
```

Comment:
I think `print_args` here could have the classic python "default args are mutable" https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil footgun, the default empty list only gets initialized once and if mutated later that mutation is present in following calls 😬

Currently that probably doesn't actually happen as we haven't noticed anything..